src/test/coins_tests.cpp

```diff
@@ -773,8 +773,10 @@ static void CheckAddCoinBase(CAmount base_value, CAmount cache_value, CAmount mo
 template <typename... Args>
 static void CheckAddCoin(Args&&... args)
 {
-    for (const CAmount base_value : {ABSENT, SPENT, VALUE1})
+    for (const CAmount base_value : {ABSENT, SPENT, VALUE1}) {
+        // NOLINTNEXTLINE(bugprone-use-after-move)
```

Comment:
As mentioned [here](https://github.com/bitcoin/bitcoin/pull/31364#discussion_r1857161138), this change struck me as dangerous.

> It shouldn't be marked nolint, rather it should be fixed: drop the forward and pass by const Args&.

I understand the desire to turn on new checks and notate the false-positives, but the result should _always_ be better/safer/cleaner code.

In this case, the result would've been a latent bug explicitly marked notabug. And while a buggy test isn't the end of the world, the suggestion that the approach for recent similar PR's might have been "quiet the static analysis tool first, fix the bug later" struck me as so backwards that I had to wonder if any other actual bugs in more important code had been masked as well. Looking through recent/current PRs, I don't see any other examples of that (modulo the boost issue pointed out here).

Please prioritize correctness and consider what you're signaling to reviewers.

Reply:
Right.

Reply:
> As mentioned [here](https://github.com/bitcoin/bitcoin/pull/31364#discussion_r1857161138), this change struck me as dangerous.

My apologies.

An additional context available here: https://github.com/bitcoin/bitcoin/pull/31306#discussion_r1846324884.